EventFilter/L1TRawToDigi/src/Block.cc

```diff
@@ -139,15 +139,31 @@ namespace l1t {
 	       ((data16[10] >> 11) != 0) || ((data16[11] >> 11) != 0)) {
          edm::LogError("L1T") << "MTF7 payload has invalid header!";
          data_ = end_;
-    } else if (
+    } else if ( // Check bits for EMTF MPC Link Errors
 	       ((data16[12] >> 15) != 0) || ((data16[13] >> 15) != 1) ||
 	       ((data16[14] >> 15) != 0) || ((data16[15] >> 15) != 0)) {
       edm::LogError("L1T") << "MTF7 payload has invalid counter block!";
       data_ = end_;
-    } else if (
-	       false) {
-      // TODO: check trailer
     }
+
+    // Check bits for EMTF Event Record Trailer, get firmware version
+    algo_ = 0; // Firmware version
```

Comment:
I think algo_ is anyway initialized with 0 in the payload constructor.

Reply:
True, but I'd rather have it explicit so that the logic for the "if (algo_ == 0)" conditional below is crystal clear.